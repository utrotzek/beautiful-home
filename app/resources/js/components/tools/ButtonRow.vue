<template>
    <div
        role="group"
        aria-label="Basic example"
        class="btn-group d-flex"
    >

        <button
            v-if="showConnect"
            type="button"
            class="btn btn-light"
            @click.stop
            @click="connect"
        >
            <i class="fas fa-link"></i> <span class="d-none d-sm-inline">Zuordnen</span>
        </button>

        <button
            v-if="showEdit"
            type="button"
            class="btn btn-light"
            @click.stop
            @click="edit"
        >
            <i class="fas fa-edit"></i> <span class="d-none d-sm-inline">Bearbeiten</span>
        </button>

        <button
            v-if="showDelete"
            type="button"
            class="btn btn-light"
            @click.stop
            @click="deleteClick"
        >
            <i class="fas fa-trash"></i> <span class="d-none d-sm-inline">Löschen</span>
        </button>

        <button
            v-if="showClose"
            type="button"
            class="btn btn-light"
            @click.stop
            @click="close"
        >
            <i class="fas fa-ban"></i> <span class="d-none d-sm-inline">Schließen</span>
        </button>
    </div>
</template>

<script>
export default {
    name: "ButtonRowVue",

    props: {
        showEdit: {
            type: Boolean,
            default: false
        },
        showDelete: {
            type: Boolean,
            default: false
        },
        showClose: {
            type: Boolean,
            default: true
        },
        showConnect: {
            type: Boolean,
            default: false
        }
    },
    methods: {
        edit() {
            this.$emit("edit");
        },
        deleteClick() {
            this.$emit("deleteClick");
        },
        close() {
            this.$emit("close");
        },
        connect() {
            this.$emit("connect");
        }
    }
};
</script>

<style scoped>

</style>
