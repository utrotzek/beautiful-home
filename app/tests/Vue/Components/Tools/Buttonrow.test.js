import {mount} from "@vue/test-utils";
import Vue from "vue";
import ButtonRow from "../../../../resources/js/components/tools/ButtonRow";

describe("Overview", () => {
    let wrapper = null;

    const allEnabled = {
        showEdit: true,
        showSave: true,
        showDelete: true,
        showClose: true,
        showConnect: true,
        showCancel: true
    };

    beforeEach(() => {
        wrapper = mount(ButtonRow);
    });

    it ("renders properly", async () => {
        wrapper.setProps(allEnabled);
        await Vue.nextTick();
        expect(wrapper.element).toMatchSnapshot();
    });

    it ("can enable edit button", async () =>  {
        expect(wrapper.find(".save").exists()).toBeFalsy();

        wrapper.setProps({showSave: true});
        await Vue.nextTick();
        expect(wrapper.find(".save").exists()).toBeTruthy();
    });

    it ("can enable delete button", async () =>  {
        expect(wrapper.find(".delete").exists()).toBeFalsy();

        wrapper.setProps({showDelete: true});
        await Vue.nextTick();
        expect(wrapper.find(".delete").exists()).toBeTruthy();
    });

    it ("can enable close button", async () =>  {
        expect(wrapper.find(".close").exists()).toBeFalsy();

        wrapper.setProps({showClose: true});
        await Vue.nextTick();
        expect(wrapper.find(".close").exists()).toBeTruthy();
    });

    it ("can enable connect button", async () =>  {
        expect(wrapper.find(".connect").exists()).toBeFalsy();

        wrapper.setProps({showConnect: true});
        await Vue.nextTick();
        expect(wrapper.find(".connect").exists()).toBeTruthy();
    });

    it ("can enable cancel button", async () =>  {
        expect(wrapper.find(".cancel").exists()).toBeFalsy();

        wrapper.setProps({showCancel: true});
        await Vue.nextTick();
        expect(wrapper.find(".cancel").exists()).toBeTruthy();
    });

    it ("buttons can be clicked", async () =>  {
        wrapper.setProps(allEnabled);
        await Vue.nextTick();

        wrapper.find(".edit").trigger("click");
        expect(wrapper.emitted().edit).toBeTruthy();

        wrapper.find(".delete").trigger("click");
        expect(wrapper.emitted().deleteClick).toBeTruthy();

        wrapper.find(".close").trigger("click");
        expect(wrapper.emitted().close).toBeTruthy();

        wrapper.find(".connect").trigger("click");
        expect(wrapper.emitted().connect).toBeTruthy();

        wrapper.find(".save").trigger("click");
        expect(wrapper.emitted().save).toBeTruthy();

        wrapper.find(".cancel").trigger("click");
        expect(wrapper.emitted().cancel).toBeTruthy();
    });


    it ("buttons can be disabled", async () =>  {
        wrapper.setProps(allEnabled);
        wrapper.setProps({disabled: true});
        await Vue.nextTick();

        wrapper.find(".edit").trigger("click");
        expect(wrapper.emitted().edit).toBeFalsy();

        wrapper.find(".delete").trigger("click");
        expect(wrapper.emitted().deleteClick).toBeFalsy();

        wrapper.find(".close").trigger("click");
        expect(wrapper.emitted().close).toBeFalsy();

        wrapper.find(".connect").trigger("click");
        expect(wrapper.emitted().connect).toBeFalsy();

        wrapper.find(".save").trigger("click");
        expect(wrapper.emitted().save).toBeFalsy();

        //cancel button should never be disabled
        wrapper.find(".cancel").trigger("click");
        expect(wrapper.emitted().cancel).toBeTruthy();
    });


});
