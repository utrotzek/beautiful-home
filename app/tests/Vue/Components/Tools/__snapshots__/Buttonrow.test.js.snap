// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders properly 1`] = `
<div
  aria-label="Basic example"
  class="btn-group d-flex"
  role="group"
>
  <button
    class="save btn btn-light"
    type="button"
  >
    <i
      class="fas fa-save"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Speichern
    </span>
  </button>
   
  <button
    class="cancel btn btn-light"
    type="button"
  >
    <i
      class="fas fa-ban"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Abbrechen
    </span>
  </button>
   
  <button
    class="connect btn btn-light"
    type="button"
  >
    <i
      class="fas fa-link"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Zuordnen
    </span>
  </button>
   
  <button
    class="edit btn btn-light"
    type="button"
  >
    <i
      class="fas fa-edit"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Bearbeiten
    </span>
  </button>
   
  <button
    class="delete btn btn-light"
    type="button"
  >
    <i
      class="fas fa-trash"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Löschen
    </span>
  </button>
   
  <button
    class="close btn btn-light"
    type="button"
  >
    <i
      class="fas fa-ban"
    />
     
    <span
      class="d-none d-sm-inline"
    >
      Schließen
    </span>
  </button>
</div>
`;
